### DISCUSSION TITLE
Urgent development needs

### DISCUSSION THREAD

**Discussion by jmellorcrummey** [HPCToolkit/hpctoolkit#599] [Ideas]
- spack recipes for intel
    - separate gtpin from level0 and opencl
- replace inter-thread communication data structures with multiple-producer, single consumer data structures
    - avoid serialization of intel instrumentation processing
- add support for threaded memory allocator (for use with inter-thread communication outside signal handler)
    - use this with new multi-producer, single consumer inter-thread communication
- improve thread view in hpcviewer
    - improve titles
    - scalable display of execution contexts
    - support both hardware and software contexts
- collect and analyze summary metrics for machine contexts, e.g. GPU, core
    - especially GPU
- suppress display of excess GPU metrics in hpcviewer by default

**Comment by Jokeren** [comment]
Hi John, can you point me a reference of the "new multi-producer, single consumer" queue you plan to implement? If events can be unordered, can we just use the triber stack?
